From c9e3656cee4a2220a53e670e4966b0e7febdb8c4 Mon Sep 17 00:00:00 2001
From: Elvis Dowson <elvis.dowson@gmail.com>
Date: Fri, 14 Dec 2012 12:18:09 +0400
Subject: [PATCH] virtex440-ml507.dts: Update device tree file using Xilinx
 ISE 14.3 with APU FPU support.

This commit updates the device tree to one generated by Xilinx ISE 14.3
- adds support for APU FPU,
- clock frequencies have been modified as a result of addinng the APU FPU,
- console output has been set to ttyUL0 with a baud rate of 115200 bps.

The bootargs have been set as follows:
console=ttyUL0 ip=on root=/dev/ram rw rootwait init=/bin/sh

This configuration is intended to be used with an initrd.image.gz
ramdisk file for the initial boot process. The size of the ramdisk file
is set to 32768 kilobytes, and is specified in the kernel defconfig file
CONFIG_BLK_DEV_RAM_SIZE=32768

Signed-off-by: Elvis Dowson <elvis.dowson@gmail.com>
---
 arch/powerpc/boot/dts/virtex440-ml507.dts |   62 ++++++++++++++---------------
 1 file changed, 31 insertions(+), 31 deletions(-)

diff --git a/arch/powerpc/boot/dts/virtex440-ml507.dts b/arch/powerpc/boot/dts/virtex440-ml507.dts
index b42a442..bdb28c0 100644
--- a/arch/powerpc/boot/dts/virtex440-ml507.dts
+++ b/arch/powerpc/boot/dts/virtex440-ml507.dts
@@ -22,9 +22,9 @@
  * MA 02111-1307 USA
  *
  * CAUTION: This file is automatically generated by libgen.
- * Version: Xilinx EDK 14.2 EDK_P.28xd
+ * Version: Xilinx EDK 14.3 EDK_P.40xd
  *
- * XPS project directory: linux_2_6_bsp_0
+ * XPS project directory: ml507_linux_2_6_bsp
  */
 
 /dts-v1/;
@@ -39,8 +39,8 @@
 		serial0 = &rs232_uart_1;
 	} ;
 	chosen {
-		bootargs = "console=tty0 ip=on root=/dev/ram rw rootwait init=/bin/sh";
-		linux,stdout-path = "/plb@0/serial@83e00000";
+		bootargs = "console=ttyUL0 ip=on root=/dev/ram rw rootwait init=/bin/sh";
+		linux,stdout-path = "/plb@0/serial@84000000";
 	} ;
 	cpus {
 		#address-cells = <1>;
@@ -49,7 +49,7 @@
 		ppc440_0: cpu@0 {
 			#address-cells = <1>;
 			#size-cells = <1>;
-			clock-frequency = <400000000>;
+			clock-frequency = <266666666>;
 			compatible = "PowerPC,440", "ibm,ppc440";
 			d-cache-line-size = <0x20>;
 			d-cache-size = <0x8000>;
@@ -60,8 +60,8 @@
 			i-cache-size = <0x8000>;
 			model = "PowerPC,440";
 			reg = <0>;
-			timebase-frequency = <400000000>;
-			xlnx,apu-control = <0x2000>;
+			timebase-frequency = <266666666>;
+			xlnx,apu-control = <0x401>;
 			xlnx,apu-udi-0 = <0x0>;
 			xlnx,apu-udi-1 = <0x0>;
 			xlnx,apu-udi-10 = <0x0>;
@@ -183,8 +183,8 @@
 			compatible = "xlnx,xps-ethernetlite-4.00.a", "xlnx,xps-ethernetlite-1.00.a";
 			device_type = "network";
 			interrupt-parent = <&xps_intc_0>;
-			interrupts = < 5 0 >;
-			local-mac-address = [ 00 0a 35 b7 78 00 ];
+			interrupts = < 8 0 >;
+			local-mac-address = [ 00 0a 35 c8 f2 00 ];
 			phy-handle = <&phy0>;
 			reg = < 0x81000000 0x10000 >;
 			xlnx,duplex = <0x1>;
@@ -219,7 +219,7 @@
 			xlnx,max-mem-width = <0x10>;
 			xlnx,mch-native-dwidth = <0x20>;
 			xlnx,mch-splb-awidth = <0x20>;
-			xlnx,mch-splb-clk-period-ps = <0x2710>;
+			xlnx,mch-splb-clk-period-ps = <0x2bf2>;
 			xlnx,mch0-accessbuf-depth = <0x10>;
 			xlnx,mch0-protocol = <0x0>;
 			xlnx,mch0-rddatabuf-depth = <0x10>;
@@ -295,9 +295,9 @@
 		iic_eeprom: i2c@81600000 {
 			compatible = "xlnx,xps-iic-2.03.a", "xlnx,xps-iic-2.00.a";
 			interrupt-parent = <&xps_intc_0>;
-			interrupts = < 7 2 >;
+			interrupts = < 9 2 >;
 			reg = < 0x81600000 0x10000 >;
-			xlnx,clk-freq = <0x5f5e100>;
+			xlnx,clk-freq = <0x54c5638>;
 			xlnx,family = "virtex5";
 			xlnx,gpo-width = <0x1>;
 			xlnx,iic-freq = <0x186a0>;
@@ -352,25 +352,25 @@
 			xlnx,tri-default = <0xffffffff>;
 			xlnx,tri-default-2 = <0xffffffff>;
 		} ;
-		rs232_uart_1: serial@83e00000 {
-			clock-frequency = <100000000>;
-			compatible = "xlnx,xps-uart16550-3.00.a", "ns16550";
-			current-speed = <9600>;
+		rs232_uart_1: serial@84000000 {
+			clock-frequency = <88888888>;
+			compatible = "xlnx,xps-uartlite-1.02.a", "xlnx,xps-uartlite-1.00.a";
+			current-speed = <115200>;
 			device_type = "serial";
 			interrupt-parent = <&xps_intc_0>;
-			interrupts = < 12 2 >;
-			reg = < 0x83e00000 0x10000 >;
-			reg-offset = <0x1003>;
-			reg-shift = <2>;
+			interrupts = < 12 0 >;
+			port-number = <0>;
+			reg = < 0x84000000 0x10000 >;
+			xlnx,baudrate = <0x1c200>;
+			xlnx,data-bits = <0x8>;
 			xlnx,family = "virtex5";
-			xlnx,has-external-rclk = <0x0>;
-			xlnx,has-external-xin = <0x0>;
-			xlnx,is-a-16550 = <0x1>;
+			xlnx,odd-parity = <0x0>;
+			xlnx,use-parity = <0x0>;
 		} ;
 		sysace_compactflash: sysace@83600000 {
 			compatible = "xlnx,xps-sysace-1.01.a", "xlnx,xps-sysace-1.00.a";
 			interrupt-parent = <&xps_intc_0>;
-			interrupts = < 4 2 >;
+			interrupts = < 7 2 >;
 			reg = < 0x83600000 0x10000 >;
 			xlnx,family = "virtex5";
 			xlnx,mem-width = <0x10>;
@@ -385,24 +385,24 @@
 			compatible = "xlnx,xps-intc-2.01.a", "xlnx,xps-intc-1.00.a";
 			interrupt-controller ;
 			reg = < 0x81800000 0x10000 >;
-			xlnx,kind-of-intr = <0x6a>;
+			xlnx,kind-of-intr = <0x1161>;
 			xlnx,num-intr-inputs = <0xd>;
 		} ;
-		xps_ps2_0: xps-ps2@81480000 {
+		xps_ps2_0: xps-ps2@86a00000 {
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "xlnx,compound";
-			ranges = < 0x0 0x81480000 0x10000 >;
+			ranges = < 0x0 0x86a00000 0x10000 >;
 			ps2@0 {
 				compatible = "xlnx,xps-ps2-1.01.b", "xlnx,xps-ps2-1.00.a";
 				interrupt-parent = <&xps_intc_0>;
-				interrupts = < 8 2 >;
+				interrupts = < 2 2 >;
 				reg = < 0x0 0x40 >;
 			} ;
 			ps2@1000 {
 				compatible = "xlnx,xps-ps2-1.01.b", "xlnx,xps-ps2-1.00.a";
 				interrupt-parent = <&xps_intc_0>;
-				interrupts = < 9 2 >;
+				interrupts = < 1 2 >;
 				reg = < 0x1000 0x40 >;
 			} ;
 		} ;
@@ -422,7 +422,7 @@
 		xps_timebase_wdt_0: xps-timebase-wdt@83a00000 {
 			compatible = "xlnx,xps-timebase-wdt-1.02.a";
 			interrupt-parent = <&xps_intc_0>;
-			interrupts = < 3 0 2 2 >;
+			interrupts = < 5 0 4 2 >;
 			reg = < 0x83a00000 0x10000 >;
 			xlnx,family = "virtex5";
 			xlnx,wdt-enable-once = <0x0>;
@@ -431,7 +431,7 @@
 		xps_timer_0: timer@83c00000 {
 			compatible = "xlnx,xps-timer-1.02.a", "xlnx,xps-timer-1.00.a";
 			interrupt-parent = <&xps_intc_0>;
-			interrupts = < 1 0 >;
+			interrupts = < 6 0 >;
 			reg = < 0x83c00000 0x10000 >;
 			xlnx,count-width = <0x20>;
 			xlnx,family = "virtex5";
-- 
1.7.10.4

